contracts/ENSGovernorV2.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./ENSGovernor.sol";
+
+/**
+ * @title ENSGovernorV2
+ * @dev Enhanced ENS Governor with bonding functionality.
+ */
+contract ENSGovernorV2 is ENSGovernor {
+    using SafeERC20 for ERC20Votes;
+
+    uint256 public forfeitedBondsBalance;
+    uint256 public lockedBondsBalance;
+
+    /**
+     * @dev ProposalBond represents the bond associated with a proposal.
+     */
+    struct ProposalBond {
+        uint256 amount; // The amount of bond deposited.
+        address proposer; // The address of the proposer who deposited the bond.
+        bool refunded; // Flag indicating if the bond has been refunded.
+        bool forfeited; // Flag indicating if the bond is forfeited.
+    }
+
+    /**
+     * @dev Mapping to store ProposalBond objects associated with proposal IDs.
+     */
+    mapping(uint256 => ProposalBond) private _proposalBonds;
+
+    /**
+     * @dev Emitted when a bond is created.
+     * @param proposer The address of the proposer.
+     * @param amount The amount of bond.
+     */
+    event BondCreated(address indexed proposer, uint256 amount);
+
+    uint256 public bondPricePerTarget = 1 ether;
+
+    /**
+     * @notice Constructor to initialize the ENSGovernorV2 contract.
+     * @param _token The token used for voting.
+     * @param _timelock The timelock controller.
+     */
+    constructor(
+        ERC20Votes _token,
+        TimelockController _timelock
+    ) ENSGovernor(_token, _timelock) {}
+
+    /**
+     * @notice Propose a governance action with a bond.
+     * @param targets The addresses to call.
+     * @param values The values to send.
+     * @param calldatas The calldata to send.
+     * @param description The description of the proposal.
+     * @return proposalId The ID of the proposal.
+     */
+    function proposeWithBond(
```

Comment:
Unless I'm missing something, the standard `propose` method is still available, correct? If proposing with a bond is optional, why would anyone do it?